client/tracing/src/logging/event_format.rs

```diff
@@ -123,7 +125,17 @@ where
 		writer: &mut dyn fmt::Write,
 		event: &Event,
 	) -> fmt::Result {
-		self.format_event_custom(CustomFmtContext::FmtContext(ctx), writer, event)
+		self.format_event_custom(CustomFmtContext::FmtContext(ctx), writer, event)?;
+		if self.dup_to_stdout && (
+			event.metadata().level() == &Level::INFO ||
+			event.metadata().level() == &Level::WARN ||
+			event.metadata().level() == &Level::ERROR
+		) {
+			let mut out = String::new();
+			self.format_event_custom(CustomFmtContext::FmtContext(ctx), &mut out, event)?;
+			print!("{}", out);
+		}
+		Ok(())
```

Comment:
```suggestion
		if self.dup_to_stdout && (
			event.metadata().level() == &Level::INFO ||
			event.metadata().level() == &Level::WARN ||
			event.metadata().level() == &Level::ERROR
		) {
			let mut out = String::new();
			self.format_event_custom(CustomFmtContext::FmtContext(ctx), &mut out, event)?;
			writer.write_str(&out)?;
			print!("{}", out);
			Ok(())
		} else {
			self.format_event_custom(CustomFmtContext::FmtContext(ctx), writer, event)
		}
```

Reply:
Depending on the costs of `format_event_custom`, we could also directly write to `stdout`.

Reply:
Unfortunately it seems like there's no common abstraction over `std::fmt::Write` and `std::io::Write`. `INFO|WARN|ERROR` messages are rare enough so that performance is not an issue here.